src/context/TodoContext.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Status } from '../types/Status';
+import { TodoContextType } from '../types/TodoContext';
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../constants/user';
+import { getTodos } from '../api/todos';
+import { Error } from '../types/Error';
+import { prepareTodos } from '../services/prepareTodos';
+
+export const TodoContext = React.createContext<TodoContextType>({
+  todos: [],
+  filteredTodos: [],
+  filterField: Status.All,
+  updatingTodosIds: [],
+  errorMessage: '',
+  tempTodo: null,
+  addTodo: () => { },
+  deleteTodo: () => { },
+  updateTodo: () => { },
+  handleFilterField: () => { },
+  handleUpdatingTodosIds: () => { },
+  handleError: () => { },
+  handleSetTempTodo: () => { },
+});
+
+type Props = {
```

Comment:
use interface prefer. 
Types have unique features but the interface is best for scaling.